Migrate Testing from Jest to Vitest (#27)

ThomasKaar opened the pull request:
# *Migrate Testing from Jest to Vitest (#27)*

## :recycle: Current situation & Problem
As described in #27, vitest is a more modern and faster testing framework.


## :gear: Release Notes 
* Added vitest
* Removed jest

## 📚 Documentation
This Pull Request migrates Jest to Vitest according to the [official vitest documentation](https://vitest.dev/guide/migration#migrating-from-jest) and the [Next.js testing guide](https://nextjs.org/docs/app/building-your-application/testing/vitest).
The packages:
* `@types/jest` 
* `jest`
* `jest-environment-jsdom`
* `ts-jest`

have been removed

## :white_check_mark: Testing
Before
```shell
$ npm run test
...
 PASS  app/page.test.tsx
  Home Component
    ✓ renders the Stanford Biodesign Digital Health Next.js Template heading (92 ms)
    ✓ renders the Stanford Biodesign Logo (6 ms)

-------------------------------|---------|----------|---------|---------|-------------------
File                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------------------|---------|----------|---------|---------|-------------------
All files                      |     100 |       75 |     100 |     100 |                   
 app                           |     100 |       75 |     100 |     100 |                   
  page.tsx                     |     100 |       75 |     100 |     100 | 19                
 packages/example-package/dist |     100 |      100 |     100 |     100 |                   
  index.js                     |     100 |      100 |     100 |     100 |                   
-------------------------------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        3.294 s
Ran all test suites.
```


After
```shell
$ npm run test
...
 ✓ app/page.test.tsx (2)
   ✓ Home Component (2)
     ✓ renders the Stanford Biodesign Digital Health Next.js Template heading
     ✓ renders the Stanford Biodesign Logo

 Test Files  1 passed (1)
      Tests  2 passed (2)
   Start at  21:47:52
   Duration  1.77s (transform 86ms, setup 0ms, collect 385ms, tests 103ms, environment 683ms, prepare 100ms)

 % Coverage report from v8
------------------------------|---------|----------|---------|---------|-------------------
File                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------------------|---------|----------|---------|---------|-------------------
All files                     |   74.46 |    66.66 |   66.66 |   74.46 |                   
 app                          |      60 |       50 |      50 |      60 |                   
  layout.tsx                  |       0 |        0 |       0 |       0 | 1-28              
  page.tsx                    |     100 |      100 |     100 |     100 |                   
 packages/example-package/src |     100 |      100 |     100 |     100 |                   
  index.ts                    |     100 |      100 |     100 |     100 |                   
------------------------------|---------|----------|---------|---------|-------------------
```


### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

ThomasKaar committed:
add migration from jest to vitest

ThomasKaar committed:
remove jest json files

PSchmiedmayer commented:
@ThomasKaar Thank you for the addition and taking a look at the issue!
I have requested a detailed review from @arkadiuszbachorski.

Looking at the PR at a high level, it seems like the coverage report you detail above is also including js files and reports a 0% test coverage for the TS files wich might not be correct.
It might make sense to take a look at some of the chances and mechanisms we made in https://github.com/StanfordBDHG/ENGAGE-HF-Web-Frontend which seems to properly report some TS test coverage despite some elements that we could improve there as well.

ThomasKaar committed:
reduce test coverage to .ts and .tsx
remove generated coverage files from eslint
add licensing to vite.config.mts

ThomasKaar committed:
fix coverage for ts files

ThomasKaar commented:
Thanks for the brief review!

I reduced the scope of the coverage analysis similar to https://github.com/StanfordBDHG/ENGAGE-HF-Web-Frontend while also including `js` files with [`sourceMap`](https://www.typescriptlang.org/tsconfig/#sourceMap) enabled so the `ts`-file can be properly mapped to the `js`-file in the coverage analysis (I updated the test plan accordingly).

Moreover, I also added licensing information to the newly added `vitest.config.mts` and ran `npm run lint:ci` locally to ensure it is working properly

PSchmiedmayer commented:
Thank you @ThomasKaar; greatly appreciated!
Just started the builds and I can sync with @arkadiuszbachorski about the next steps 👍
